src/core/counter.h

```diff
@@ -0,0 +1 @@
+struct counter { uint64_t c; };
```

Comment:
`link.h` uses `struct counter`. Is that what you mean by "for the benefit of C files"?